File: pog-store/core/src/main/java/br/com/pogstore/core/common/BaseRequest.java
package br.com.pogstore.core.common;

import java.util.UUID;

public abstract class BaseRequest {

	private String identifier;
	
	protected BaseRequest() {
		this.identifier = UUID.randomUUID().toString();
	}

	protected BaseRequest(String identifier) {
		this.identifier = identifier;
	}

	protected BaseRequest(UUID identifier){
		this(identifier.toString());
	}
	
	public String getIdentifier() {
		return identifier;
	}
	
}

File: cqrs/crime-map/src/CrimeMap.Common/Messages/BaseRequest.cs
using System;
using System.Runtime.Serialization;

namespace CrimeMap.Common {

	[DataContract]
	public abstract class BaseRequest {

		protected BaseRequest() {
			Identifier = Guid.NewGuid().ToString();
		}

		protected BaseRequest(string idetifier) {
			Identifier = Identifier;
		}

		protected BaseRequest(Guid identifier) : this(identifier.ToString()) { }

		[DataMember]
		public string Identifier { get; protected set; }

	}
}
